packages/core/react/use-assistant.ts

```diff
@@ -0,0 +1,105 @@
+import { nanoid } from 'nanoid';
+import { useState } from 'react';
+import { processMessageStream } from '../shared/process-message-stream';
+import { Message } from '../shared/types';
+
+export type AssistantStatus = 'in_progress' | 'awaiting_message';
+
+export function useAssistant_experimental({
```

Comment:
should this be `experimental_useAssistant`?

Reply:
That's what I used initially - it's prevented by the hooks linting rule.